packages/dds/tree/src/test/util/testTreeProvider.spec.ts

```diff
@@ -8,18 +8,53 @@ import { SharedTreeCore } from "../../shared-tree-core";
 import { spyOnMethod, TestTreeProvider } from "../utils";
 
 describe("TestTreeProvider", () => {
-    it("can manually trigger summaries", async () => {
+    it("can manually trigger summaries with summarizeOnDemand", async () => {
         let summaryCount = 0;
+        const unspy = spyOnMethod(SharedTreeCore, "summarizeCore", () => {
+            summaryCount += 1;
+        });
+
+        const provider = await TestTreeProvider.create(1, true);
+        const summaries = summaryCount;
+        await provider.summarize();
+
+        assert(summaryCount === summaries + 1);
+        unspy();
+    });
+
+    it("cannot manually trigger summaries without setting summarizeOnDemand", async () => {
+        let summarizerError;
+        try {
+            const provider = await TestTreeProvider.create(1);
+            await provider.summarize();
+        } catch (error) {
+            summarizerError = error;
+        }
+        assert(summarizerError !== undefined);
```

Comment:
I tried to use `assert.throws`, but it seems to ask me to provide the expected error object to be thrown. I tried to manually create it by calling the `AssertionError()` function, but this did not seem to work. The parameters I passed in were the fields I found for the error object in the `try` and `catch` statement from the current test.

Do you happen to know if there is a better way to get the exact `AssertionError` object?

My attempt:
```
        const summarizerError = new AssertionError({
            actual: false,
            expected: true,
            operator:'==',
            message: "can't summarize, because must create at least one tree."
        });
        const provider = await TestTreeProvider.create(1);
        assert.throws(
            async () => provider.summarize(),
            summarizerError
        )
```

Reply:
You don't have to provide a second argument if you don't want. If you want to check that it's the right kind of error though, search our packages for `assert.throws` and you'll see some examples, e.g. the one in utilities.spec.ts